autopts/rtt.py

```diff
@@ -233,9 +243,23 @@ def start(self, buffer_name, log_filename, device_core, debugger_snr):
         self.log_file = open(log_filename, 'ab')
         self.rtt_reader.start(buffer_name, device_core, debugger_snr, self._on_line_read_callback, (self.log_file,))
 
-    def stop(self):
+    def stop_timer(self):
+        # timeout reached
+        self.rtt_reader.rtt_stop = True
+
+    def stop(self, reset, timeout=0.1):
         log("%s.%s", self.__class__, self.stop.__name__)
-        self.rtt_reader.stop()
+
+        if reset:
+            self.rtt_reader.stop()
```

Comment:
Should this wait even when when this is true?

Reply:
This could be implemented, but from what I understood the rtt closing issue occured at the end of the test case which is why I omitted this case. Ideally IUT should be reseted only at the beginning of test case, If you encounter unexpected IUT reset while in the middle/end of test case please let me know.